libparsers/src/xml/route_marker.cpp

```diff
@@ -158,8 +158,16 @@ namespace xml {
 			float parse_distance(rapidxml_ns::xml_node<char>* distance_node,
 			                     errors::errors_t& errors) {
 				try {
-					return util::parse_loose_float(
+
+					auto const time = util::parse_loose_float(
 					    std::string(distance_node->value(), distance_node->value_size()));
+					if(time < 0){
```

Comment:
I was hilariously wrong about this, it can be negative, I'll fix it though as this code looks good as is.